tailscale/resource_device_subnet_routes.go

```diff
@@ -27,7 +27,17 @@ func resourceDeviceSubnetRoutes() *schema.Resource {
 		UpdateContext: resourceDeviceSubnetRoutesUpdate,
 		DeleteContext: resourceDeviceSubnetRoutesDelete,
 		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+				// We can't do a simple passthrough here as the ID used for this resource is a
+				// randomly generated UUID and we need to instead fetch based on the device_id.
+				//
+				// TODO(mpminardi): investigate changing the ID in state to be the device_id instead
+				// in an eventual major version bump.
+				d.Set("device_id", d.Id())
+				d.SetId(createUUID())
```

Comment:
Hey @reegnz!

The rest of the `device*` resources are using `device_id` as the `id` in state upon resource creation (and were before the import changes), so in those cases the passthrough can be used and works as expected ([device_key](https://github.com/tailscale/terraform-provider-tailscale/blob/main/tailscale/resource_device_key.go#L54), [device_tags](https://github.com/tailscale/terraform-provider-tailscale/blob/main/tailscale/resource_device_tags.go#L82), and [device_authorization](https://github.com/tailscale/terraform-provider-tailscale/blob/main/tailscale/resource_device_authorization.go#L66) for reference).

For this resource the change [here](https://github.com/tailscale/terraform-provider-tailscale/pull/442/files#diff-8248a0e44d32b274a01068ad5dae1501f8a3611e5958baaf94c65143671fffb2R52) would end up breaking existing users of the resource as the Id of existing `device_subnet` resources are a UUID in state, meaning that we would try and fail to fetch the device using the UUID instead of the device Id for non-imported resources.

This is why we need to do the more complex import logic in this case: so that we can properly read both when doing the import and when doing a read operation for an existing resource. Doing a fresh import would not have surfaced this issue as the read would be using the passthrough Id (e.g., the `device_id`) and work properly for that scenario. 

This resource _should_ also use the `device_id` as its id, but since we have released versions where it uses a UUID instead we need to keep this in UUID format to stay within what we can release without doing a major version revision as per the [HashiCorp versioning best practices](https://developer.hashicorp.com/terraform/plugin/best-practices/versioning#example-major-number-increments).

Apologies as well for not catching this on the original review! I had forgotten that this resource was using a UUID for its Id and completely missed that this would cause issues originally.